include/npcomp/Dialect/Refbackrt/IR/RefbackrtOps.td

```diff
@@ -68,12 +68,23 @@ def Refbackrt_FuncMetadataOp
   let description = [{
     Runtime metadata for a single func.
 
-    TODO: Augment this with information for type/shape checking of arguments.
+    TODO: Augment this with type/shape checking of arguments.
+    // Need to figure out how we want to represent the argType's here.
   }];
   let arguments = (ins
     FlatSymbolRefAttr:$funcName,
     I32Attr:$numInputs,
-    I32Attr:$numOutputs
+    I32Attr:$numOutputs,
+    OptionalAttr<I32ElementsAttr>:$inputTypes,
```

Comment:
These confuse me. Can you document the semantics of each ElementsAttr?

In ptaricular, I find it strange that inputShapes is an elementsAttr -- in general, different inputs can have different ranks, which makes the inputShapes nested array non-hyperrectangular and so not representable with an ElementsAttr.

Reply:
Documentation added:

https://github.com/brycearden/mlir-npcomp/blob/0ae8218f2370c1be90df2f3523d5018223c6a096/include/npcomp/Dialect/Refbackrt/IR/RefbackrtOps.td#L65-L117